### ISSUE TITLE
Issues with Bundler 2.4.16 Cache & Write Only Filesystems

### ISSUE THREAD

**Issue by metaskills** [rubygems/rubygems#6813]
### Describe the problem as clearly as you can

During some recent container deployments we hit an issue where Bundler changed its behavior with regard to the cache. 

This may be off the beaten path, but we remove the bundler path since we are cloning a lot of repos during our Bundle install and said repos are rather large. For example prior to doing a Docker `COPY . .` we do this in our build process.

```
rm -rf vendor/bundle/ruby/*/cache
```

Prior to Bundler v2.4.16 this was just fine. I can show you an example of a before and after below. The fact that `bundler/setup` now writes to disk was the source of our issue and kept our application from booting.

2.4.15

```
rm -rf vendor/bundle/ruby/*/cache
BUNDLE_GEMFILE=/aoo/Gemfile ruby -e "require 'bundler/setup'"
> tree vendor/bundle/ruby/3.1.0/cache
vendor/bundle/ruby/2.7.0/cache [error opening dir]
0 directories, 0 files
```

2.4.16

```
rm -rf vendor/bundle/ruby/*/cache
BUNDLE_GEMFILE=/app/Gemfile ruby -e "require 'bundler/setup'"
tree vendor/bundle/ruby/3.1.0/cache
vendor/bundle/ruby/2.7.0/cache
└── bundler
    └── git
```

### Did you try upgrading rubygems & bundler?

Yes, that is the issue. When we build our containers, we always install the latest version of ruby gems and bundler.

### Post steps to reproduce the problem

See description above and usage of `bundler/setup`.

### What were you expecting to happen?

I was expecting that Bundler would not try to write to the file system when the cache was removed. 

### What actually happened?

It created an empty `bundler/git` directory in the vendor cache.

**Comment by deivid-rodriguez** [comment]
Thanks for the report, let me have a look!

**Comment by metaskills** [comment]
Thanks... what is the release cycle for Bundler? Will this go out in a few days or so?

**Comment by deivid-rodriguez** [comment]
We normally release every 2 weeks, but since this was a regression, I released 2.4.17 to fix it.
